src/snowflake/snowpark/modin/plugin/_internal/indexing_utils.py

```diff
@@ -478,34 +478,53 @@ def get_frame_by_row_pos_slice_frame(
     # Row position column required for left and right bound comparison.
     frame = internal_frame.ensure_row_position_column()
     row_pos_col = col(frame.row_position_snowflake_quoted_identifier)
-
-    # create a count column which will be used to check step
-    frame = frame.append_column("count", pandas_lit(1) + max_(row_pos_col).over())
-    count_col = col(frame.data_column_snowflake_quoted_identifiers[-1])
+    count_col_created = False
```

Comment:
we have added ensure_row_count_column to help append a row count column to a dataframe and avoid repeated row count column creation, let's reuse that